# Amazon Sales — Business Metrics

Este notebook tem como objetivo criar e consolidar métricas de negócio
a partir dos dados tratados, que serão utilizadas no dashboard do Power BI
e no acompanhamento da performance de vendas.


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../DB/amazon_sales_data_2025.csv')
df['order_date'] = pd.to_datetime(df['order_date'])

In [10]:
print('\n📊 MAIN BUSINESS METRICS\n')

print(f'💰 Total Revenue: ${total_revenue:,.2f}')
print(f'🧾 Total Orders: {total_orders}')
print(f'📦 Total Quantity Sold: {total_quantity}')
print(f'🎯 Average Ticket: ${average_ticket:,.2f}')




📊 MAIN BUSINESS METRICS

💰 Total Revenue: $243,845.00
🧾 Total Orders: 250
📦 Total Quantity Sold: 714
🎯 Average Ticket: $975.38


As métricas principais indicam um total de 250 pedidos, com receita acumulada
de aproximadamente $243 mil e ticket médio em torno de $975 por pedido.


In [16]:
monthly_metrics = (
    df.set_index('order_date')
      .resample('ME')
      .agg(
          revenue=('revenue', 'sum'),
          orders=('order_id', 'nunique'),
          quantity=('quantity', 'sum')
      )
      .reset_index()
)
print('\n📆 MONTHLY METRICS\n')

for index, row in monthly_metrics.iterrows():
    print(
        f"{row['order_date'].strftime('%Y-%m')} | "
        f"Revenue: ${row['revenue']:,.2f} | "
        f"Orders: {row['orders']} | "
        f"Quantity: {row['quantity']}"
    )



📆 MONTHLY METRICS

2025-01 | Revenue: $2,120.00 | Orders: 4 | Quantity: 7
2025-02 | Revenue: $77,610.00 | Orders: 80 | Quantity: 232
2025-03 | Revenue: $78,295.00 | Orders: 94 | Quantity: 265
2025-04 | Revenue: $12,715.00 | Orders: 6 | Quantity: 18
2025-05 | Revenue: $8,300.00 | Orders: 7 | Quantity: 25
2025-06 | Revenue: $21,120.00 | Orders: 11 | Quantity: 33
2025-07 | Revenue: $11,180.00 | Orders: 7 | Quantity: 20
2025-08 | Revenue: $4,765.00 | Orders: 9 | Quantity: 29
2025-09 | Revenue: $3,750.00 | Orders: 5 | Quantity: 11
2025-10 | Revenue: $14,930.00 | Orders: 14 | Quantity: 41
2025-11 | Revenue: $4,790.00 | Orders: 6 | Quantity: 17
2025-12 | Revenue: $4,270.00 | Orders: 7 | Quantity: 16


In [17]:
city_metrics = (
    df.groupby('customer_location')
      .agg(
          revenue=('revenue', 'sum'),
          orders=('order_id', 'nunique')
      )
      .sort_values(by='revenue', ascending=False)
      .reset_index()
)
print('\n🌍 METRICS BY CITY\n')

for index, row in city_metrics.iterrows():
    print(
        f"{row['customer_location']} | "
        f"Revenue: ${row['revenue']:,.2f} | "
        f"Orders: {row['orders']}"
    )



🌍 METRICS BY CITY

Miami | Revenue: $31,700.00 | Orders: 31
Denver | Revenue: $29,785.00 | Orders: 26
Houston | Revenue: $28,390.00 | Orders: 32
Dallas | Revenue: $27,145.00 | Orders: 29
Seattle | Revenue: $26,890.00 | Orders: 22
Boston | Revenue: $26,170.00 | Orders: 26
Chicago | Revenue: $20,810.00 | Orders: 22
New York | Revenue: $18,940.00 | Orders: 26
Los Angeles | Revenue: $17,820.00 | Orders: 17
San Francisco | Revenue: $16,195.00 | Orders: 19


In [18]:
category_metrics = (
    df.groupby('category')
      .agg(
          revenue=('revenue', 'sum'),
          orders=('order_id', 'nunique'),
          quantity=('quantity', 'sum')
      )
      .sort_values(by='revenue', ascending=False)
      .reset_index()
)
print('\n🧺 METRICS BY CATEGORY\n')

for index, row in category_metrics.iterrows():
    print(
        f"{row['category']} | "
        f"Revenue: ${row['revenue']:,.2f} | "
        f"Orders: {row['orders']} | "
        f"Quantity: {row['quantity']}"
    )



🧺 METRICS BY CATEGORY

Electronics | Revenue: $129,950.00 | Orders: 118 | Quantity: 348
Home Appliances | Revenue: $105,000.00 | Orders: 40 | Quantity: 110
Footwear | Revenue: $4,320.00 | Orders: 27 | Quantity: 72
Clothing | Revenue: $3,540.00 | Orders: 40 | Quantity: 115
Books | Revenue: $1,035.00 | Orders: 25 | Quantity: 69


In [19]:
payment_metrics = (
    df.groupby('payment_method')
      .agg(
          revenue=('revenue', 'sum'),
          orders=('order_id', 'nunique')
      )
      .sort_values(by='orders', ascending=False)
      .reset_index()
)

print('\n💳 METRICS BY PAYMENT METHOD\n')

for index, row in payment_metrics.iterrows():
    print(
        f"{row['payment_method']} | "
        f"Revenue: ${row['revenue']:,.2f} | "
        f"Orders: {row['orders']}"
    )




💳 METRICS BY PAYMENT METHOD

PayPal | Revenue: $69,645.00 | Orders: 60
Credit Card | Revenue: $61,595.00 | Orders: 54
Debit Card | Revenue: $31,900.00 | Orders: 53
Gift Card | Revenue: $47,955.00 | Orders: 42
Amazon Pay | Revenue: $32,750.00 | Orders: 41


## Conclusão das Métricas de Negócio

As métricas foram apresentadas de forma clara e padronizada,
utilizando saídas formatadas em texto para facilitar a leitura
e interpretação dos principais indicadores de desempenho.

Essa abordagem garante entendimento rápido dos resultados e
servirá de base para a construção dos KPIs no dashboard do Power BI.

